frontend/libs/studio-components/src/components/StudioTextField/StudioTextfieldSchema/StudioTextfieldSchema.test.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { StudioJSONValidatorUtils } from '../StudioJSONValidatorUtils';
+import { StudioTextfieldSchema, type StudioTextfieldSchemaProps } from './StudioTextfieldSchema';
+import type { JsonSchema } from '../../../../../../packages/shared/src/types/JsonSchema';
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+const handleOnChange = jest.fn();
+
+describe('StudioTextfieldSchema', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render StudioTextfieldSchema', async () => {
+    renderStudioTextfieldSchema();
+    expect(screen.getByText('test')).toBeInTheDocument();
+  });
+
+  it('should call handleOnchange when changing text input', async () => {
+    const user = userEvent.setup();
+    renderStudioTextfieldSchema({
+      inputProps: { onChange: handleOnChange, icon: <div>icon</div> },
+    });
+    const editComponentIdButton = screen.getByRole('button', { name: /test/i });
+    expect(editComponentIdButton).toBeInTheDocument();
+    await act(() => user.click(editComponentIdButton));
+    const input = screen.getByRole('textbox');
+    await act(() => user.type(input, 'test'));
+    expect(handleOnChange).toHaveBeenCalled();
+  });
+
+  it('should call validateAgainstSchema when changing text input', async () => {
+    const user = userEvent.setup();
+    const spyValidateAgainstSchema = jest.spyOn(
+      StudioJSONValidatorUtils.prototype,
+      'validateProperty',
+    );
```

Comment:
How about sending a validator method into jsonValidator instead and checking if it gets called as expected? 😊 This way, we avoid the need for a spyOn here.